File: 724-find-pivot-index/find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        n = len(nums)

        prefix = [0] * (n + 1)
        for i in range(len(nums)):
            prefix[i + 1] = prefix[i] + nums[i]
        
        for i in range(n):
            left = prefix[i]
            right = prefix[-1] - prefix[i + 1]
            if left == right:
                return i
        return -1

File: 724-find-pivot-index/find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int n = nums.size();
        int left_sum = 0;
        int total = accumulate(nums.begin(), nums.end(), 0);

        for (int i = 0; i < n; ++i) {
            int right_sum = total - nums[i] - left_sum;
            if (right_sum == left_sum) {
                return i;
            }
            left_sum += nums[i];
        }
        return -1;
        
    }
};